ark/phenotyping/som_utils.py

```diff
@@ -737,10 +738,79 @@ def pixel_consensus_cluster(fovs, channels, base_dir, max_k=20, cap=3,
             print(output.strip())
 
 
-def visualize_pixel_cluster_data(fovs, channels, base_dir, cluster_dir,
-                                 pixel_cluster_col='cluster', dpi=None, center_val=None,
-                                 overlay_values=False, colormap="vlag",
-                                 save_dir=None, save_file=None):
+def visualize_pixel_cluster_counts(fovs, channels, base_dir, data_dir,
+                                   pixel_cluster_col='cluster', figsize=(50, 20),
+                                   color='#00FF00', dpi=None,
+                                   save_dir=None, save_file=None):
+    """Visualize the number of pixels per cluster in a bar chart
+
+    Args:
+        fovs (list):
+            The list of fovs to subset on
+        channels (list):
+            The list of channels to subset on
+        base_dir (str):
+            The path to the data directories
+        data_dir (str):
+            The path to the data directory, either the cluster or consensus dir
+        pixel_cluster_col (str):
+            The name of the cluster column to visualize, should be 'cluster' or 'hCluster_cap'
+        figsize (tuple):
+            A tuple determining the x and y dimension of the figure to plot
+        color (str):
+            The color of the bars in the barchart
+        dpi (float):
+            The resolution of the image to save, ignored if save_dir is None
+        save_dir (str):
+            If specified, a directory where we will save the plot
+        save_file (str):
+            If save_dir specified, specify a file name you wish to save to.
+            Ignored if save_dir is None
+    """
+
+    # verify the pixel_cluster_col provided is valid
+    misc_utils.verify_in_list(
+        provided_cluster_col=pixel_cluster_col,
+        valid_cluster_cols=['cluster', 'hCluster_cap']
+    )
+
+    # compute the channel average dataframe with counts
+    cluster_avgs = compute_pixel_cluster_channel_avg(fovs, channels, base_dir,
+                                                     pixel_cluster_col, data_dir,
+                                                     keep_count=True)
+
+    # convert the cluster column to integer type
+    cluster_avgs[pixel_cluster_col] = cluster_avgs[pixel_cluster_col].astype(int)
+
+    # keep just the pixel_cluster_col and the count column
+    cluster_avgs = cluster_avgs[[pixel_cluster_col, 'count']]
+
+    # need to make pixel_cluster_col string type so it displays properly (also int, no decimals)
+    cluster_avgs[pixel_cluster_col] = cluster_avgs[pixel_cluster_col].astype(int).astype(str)
+
+    # define a custom title
+    if pixel_cluster_col == 'cluster':
+        title = 'Distribution of pixel SOM cluster counts'
+        x_label = 'Pixel SOM cluster'
+    else:
+        title = 'Distribution of pixel meta cluster counts'
+        x_label = 'Pixel meta cluster'
+
+    y_label = 'Count'
```

Comment:
a `kwargs` pass-down also lets people set these without busying up the parameter/argument list.

Reply:
I think we could change this so the data preprocessing is in the function, and then we just call the boxplot function direclty in the cell so that people can manipulate it how they want.